pkg/reconciler/pipeline/controller.go

```diff
@@ -942,3 +945,52 @@ func (r *pipelineReconciler) safeToDelete(ctx context.Context, pl *dfv1.Pipeline
 	}()
 	return daemonClient.IsDrained(ctx, pl.Name)
 }
+
+func checkChildrenResourceStatus(ctx context.Context, c client.Client, pipeline *dfv1.Pipeline) error {
+	// get the daemon deployment and update the status of it to the pipeline
+	var daemonDeployment appv1.Deployment
+	if err := c.Get(ctx, client.ObjectKey{Namespace: pipeline.GetNamespace(), Name: pipeline.GetDaemonDeploymentName()},
+		&daemonDeployment); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Shall we add `apierrors.IsNotFound(err)`? that means daemon service does not exist.

Reply:
When returning error, we should also mark unhealthy with reason.

Reply:
Just after creating daemon deployment, it's not able to find it instant and was giving error here, so have added check `!apierrors.IsNotFound(err)`

Reply:
hmmm.. Logically, skipping it is not right, maybe put a word like "Deployment not found, might be still under creation" when no found?

Reply:
Do we need to reconcile also for `apierrors.IsNotFound(err)` error as returning `err`, I have checked other places it's not reconciling for `apierrors.IsNotFound(err)`?